### ISSUE TITLE
HeaderButtons that look like links do not behave as links

### ISSUE THREAD

**Issue by brdunfield** [veteransaffairscanada/vac-benefits-directory#1351]
For example, the user cannot right click and open them in a new tab, the URL of the link is not visible in the browser before clicking, etc.

This is because these elements are styled buttons. We should probably find a way to make them anchors, most critically links that exit the app such as the MyVAC link in the favourites page.

There may be issues around Redux and maintaining data if "internal" links are changed to anchors.

**Comment by smcmurtry** [comment]
I just noticed that this issue also exists for the title of embedded benefit cards, so I opened #1359

**Comment by brdunfield** [comment]
I think this issue might need other accessibility / design decisions to be made, and when implemented will likely split HeaderButton into two components - one for true buttons (such as "Save for Later" on the benefits list which is also using HeaderButton and should actually be styled as a button for accessibility), and one for links as they are being used on the Favourites Page.

**Comment by brdunfield** [comment]
Leads into #1388
